### ISSUE TITLE
4.4 Inspect Fields

### ISSUE THREAD

**Issue by IanMayo** [debrief/pepys-import#767]
## 🐞 Overview

We're going to let analysts edit one or more rows of data at a time.

The edit form UI will be one row per attribute, in two columns.  The left-hand column is _Current_ value, and the right-hand column is the _new_ value.  Only attributes with values in the right-hand row get edited.  This is how we can, for example, select 5 platforms, and change them all to `France` nationality.

This task is to inspect the selected data, and collate the data necessary to populate the edit form.

## 🔗 Feature

#600  Maintenance Tools

## API
Receives: one or more IDs for a database table
Processing: use inspection to collate list of attributes & editing data
Returns: structure containing:
* Data table/type name (`Platforms`, `Measurements`)
* Array of attributes, containing:
    - name (human-readable)
    - name (database attribute name)
    - data type, one of:
        - ID (not editable)
        - Number
        - List (with pairs of ID & human-readable name)
        - Text
    - array containing current value (or current values if number of IDs passed in)

## 🔢 Acceptance criteria

* Pass in a platform ID
* Get a structure detailing the attributes of a platform
* Pass in several State IDs
* Get a structure detailing attributes of these States
* Pass in a nationality ID
* Get a structure detailing the nationality

**Comment by robintw** [comment]
I'm not properly working today, but just to say that I've already had to implement quite a bit of this to get the column data structure for the FilterWidget. I'll post a link here to that code tomorrow.

We'll also have to think about how we design the UI if we're editing multiple rows at once - and how we deal with new values for foreign keyed columns (eg. If the user wants to change the platform type, do we get them to select a new platform type from a drop-down list? What if they want to edit it to a new platform type - do they have to use a different function to create that platform type first?)

Anyway, more on this tomorrow hopefully

**Comment by IanMayo** [comment]
Thanks for the feedback @robintw 

Maybe the solution is to do some refactoring, to pull out common blocks of code (and move it away from UI code, if that's applicable).

@clenoble will be developing the mockup to show how an Editor Popup can edit one or multiple rows.  She may not get that done until next week, but I don't see that being an issue.

**Comment by robintw** [comment]
Having thought about this a bit more: we'll already have access to most of this data from the column_data data structure that we have stored as an attribute of the MaintenanceGUI class (we had to create it for use in the FilterWidget). Loading it all from the database again would be a waste of time, so I suggest that we use a combination of the already-existing column data, plus a few new bits of data that we need to collect for editing (eg. the current values of the fields).

I can start putting this together fairly easily once I have a bit more of an idea about how the edit functionality will be working.
